### PR TITLE
Add some concepts information to the API toolbar

### PR DESCRIPTION
**Pull request by alexwlchan**
## Who is this for?

Me

## What is it doing for them?

Making it easier for us to follow what's happening on the concepts pages. In particular, I now get:

* A link to the concepts API JSON
* A list of identifiers on the concept

Because it's hard to follow discussions about the source of the concepts data when I can't see where the data is coming from.

Screenshot 2022-09-27 at 19 17 03

## Implementation note

This continues a pattern I started in #8491, where individual pages know what their API toolbar links should be, rather than having the API toolbar hold all that context, and potentially have to refetch data that's already available on the page.

e.g. we already know what the concept data is, so we reuse that to build the toolbar rather than querying the concepts API a second time.

This is slightly more efficient and avoids bloating the API toolbar code.

### CODE REVIEW

File: catalogue/webapp/pages/concept.tsx [hunk 1]
```diff
@@ -393,17 +447,6 @@ export const getServerSideProps: GetServerSideProps<Props | AppErrorProps> =
...
-    if (conceptResponse.type === 'Error') {
-      if (conceptResponse.httpStatus === 404) {
-        return { notFound: true };
-      }
-      return appError(
-        context,
-        conceptResponse.httpStatus,
-        conceptResponse.description
-      );
-    }
```
Comment (alexwlchan) [inline catalogue/webapp/pages/concept.tsx:405]: This is an unrelated improvement I spotted while reading this code – if the concept lookup fails, there's no point waiting to see if we can get related works/images; we might as well fail immediately.

File: catalogue/webapp/pages/concept.tsx [hunk 2]
```diff
@@ -335,6 +340,44 @@ export const ConceptPage: NextPage<Props> = ({
...
 };
 
+function getDisplayIdentifierType(identifierType: IdentifierType): string {
+  switch (identifierType.id) {
+    case 'lc-names':
+      return 'LC Names';
+    case 'lc-subjects':
+      return 'LCSH';
+    case 'nlm-mesh':
+      return 'MeSH';
```
Comment (alexwlchan) [inline catalogue/webapp/pages/concept.tsx:350]: Yes, lowercase 'e' is the correct capitalisation. https://www.nlm.nih.gov/mesh/meshhome.html

**Comment by rcantin-w** [comment]
Looks good 👍 :) 
Only question is; should/could it be moved to `ApiToolbar` like the logic for the Tzitzit is?

**Comment by alexwlchan** [comment]
> should/could it be moved to `ApiToolbar` like the logic for the Tzitzit is?

could: yes; should: no

Or at least, that's what I think – see implementation note. The API toolbar gets loaded on every page, and I'm wary of bloating it up with page-specific pieces. I think the API toolbar links should live on the pages instead. I'm not about to move all the existing links, but I'm trying to be better going forward.

**Comment by rcantin-w** [comment]
> see implementation note.

Indeed! Should've scrolled past the image 😅 It makes a lot of sense - thanks for that 👍

**Review by rcantin-w — APPROVED** [review]
🙌
